File: 0907-koko-eating-bananas/solution.py
class Solution:
    def totalh(self,piles,k):
        total=0
        for i in range(len(piles)):
            total=total+math.ceil(piles[i]/k)
        return total
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        start=1
        end=max(piles)
        while(start<=end):
            mid=start+(end-start)//2
            if(self.totalh(piles,mid)<=h):
                end=mid-1
            else:
                start=mid+1
        return start

        

File: 0907-koko-eating-bananas/solution.java
class Solution {
    public long findhours(int[] piles, int k){

    long hours = 0;

    for(int i = 0; i < piles.length; i++){
        long hour = (long)Math.ceil((double)piles[i] / k);
        hours += hour;
    }

    return hours;
}
    public int minEatingSpeed(int[] piles, int h) {
        int low=1;
        int high=0;
        for(int i=0;i<piles.length;i++){
            high=Math.max(high,piles[i]);
        }
        while(low<=high){
            int mid=low+(high-low)/2;
            long hr=findhours(piles,mid);
            if(hr<=h){
                high=mid-1;
            }
            else{
                low=mid+1;
            }
        }
        return low;
    }
}
